Frontend implementation: Login, SignUp, Start, and HUD components

Bryang2303 opened the pull request:
This PR implements the main frontend components for the project:
- Login component
- SignUp component
- Start screen
- HUD component

Closes #2 
Closes #3 
Closes #4 
Closes #5 
Closes #13

Bryang2303 committed:
Front end project creation and assets addition

Project created with Vite JS, Tailwind CSS installed and some images needed for the GUI added.

Bryang2303 committed:
Start component creation

Component Start added and personalized

Bryang2303 committed:
Login Component creation

Component Login added and personalized

Bryang2303 committed:
SignUp Component creation

Component SignUp added and personalized

Bryang2303 committed:
HUD Component creation

Component HUD added and personalized (remote control of the car and status view)

Bryang2303 committed:
Some route changes

Bryang2303 committed:
Some changes for private routes

Private route functionality added to the HUD. To avoid the control without being logged.

AlpizarGregory commented on project_1/front_end_car/src/components/Login.jsx:
```diff
@@ -0,0 +1,82 @@
+        return;
+      }
+
+      // Guardar usuario en contexto
```
Remove trash code, like this comment-out code and so. 
If some commented'out code needs to be kept, add a note that explains what it does and why it is commented.

AlpizarGregory commented on project_1/front_end_car/src/components/HUD.jsx:
```diff
@@ -0,0 +1,257 @@
+function HUD() {
+  const { currentUser } = useContext(MainContext);
+
+  const [speed, setSpeed] = useState(50); // valor inicial en 50%
```
Try to use just english, translate your comments

Bryang2303 committed:
Log out proper system and changes in comments

Translation to only english applied to the comments of the code and a log out option added.
